app/feature/feature-delete-account/src/main/kotlin/com/hedvig/android/feature/deleteaccount/DeleteAccountDestination.kt

```diff
@@ -0,0 +1,199 @@
+import hedvig.resources.R
+
+@Composable
+internal fun DeleteAccountDestination(
+  viewModel: DeleteAccountViewModel,
+  navigateUp: () -> Unit,
+  navigateBack: () -> Unit,
+) {
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+  DeleteAccountScreen(
+    uiState = uiState,
+    navigateUp = navigateUp,
+    navigateBack = navigateBack,
+    retryLoading = { viewModel.emit(DeleteAccountEvent.RetryLoading) },
+    initiateAccountDeletion = { viewModel.emit(DeleteAccountEvent.InitiateAccountDeletion) },
+  )
+}
+
+@Composable
+private fun DeleteAccountScreen(
+  uiState: DeleteAccountUiState,
+  navigateUp: () -> Unit,
+  navigateBack: () -> Unit,
+  retryLoading: () -> Unit,
+  initiateAccountDeletion: () -> Unit,
+) {
+  HedvigScaffold(navigateUp = navigateUp) {
+    when (uiState) {
+      DeleteAccountUiState.FailedToLoadDeleteAccountState -> {
+        HedvigErrorSection(retry = retryLoading, Modifier.weight(1f))
+      }
+
+      DeleteAccountUiState.Loading -> {
+        HedvigFullScreenCenterAlignedProgress(Modifier.weight(1f))
+      }
+
+      is DeleteAccountUiState.Success -> {
+        when (uiState) {
+          is DeleteAccountUiState.Success.CanDelete -> {
+            if (uiState.failedToPerformDeletion) {
+              HedvigErrorSection(retry = retryLoading, Modifier.weight(1f))
+            } else {
+              DeleteScreenContents(
+                title = stringResource(R.string.DELETE_ACCOUNT_DELETE_ACCOUNT_TITLE),
+                description = stringResource(R.string.DELETE_ACCOUNT_DELETE_ACCOUNT_DESCRIPTION),
+                buttonText = stringResource(R.string.PROFILE_DELETE_ACCOUNT_CONFIRM_DELETION),
+                onButtonClick = initiateAccountDeletion,
+                modifier = Modifier.weight(1f),
+                isButtonLoading = uiState.isPerformingDeletion,
+                buttonColors = ButtonDefaults.buttonColors(
+                  containerColor = MaterialTheme.colorScheme.error,
+                  contentColor = MaterialTheme.colorScheme.onError,
+                ),
+              )
+            }
+          }
+          DeleteAccountUiState.Success.AlreadyRequestedDeletion,
+          DeleteAccountUiState.Success.HasActiveInsurance,
+          DeleteAccountUiState.Success.HasOngoingClaim,
+          -> {
+            DeleteScreenContents(
+              title = when (uiState) {
+                DeleteAccountUiState.Success.AlreadyRequestedDeletion -> stringResource(
+                  R.string.DELETE_ACCOUNT_PROCESSED_TITLE,
+                )
+                DeleteAccountUiState.Success.HasActiveInsurance -> stringResource(
+                  R.string.DELETE_ACCOUNT_YOU_HAVE_ACTIVE_INSURANCE_TITLE,
+                )
+                DeleteAccountUiState.Success.HasOngoingClaim -> stringResource(
+                  R.string.DELETE_ACCOUNT_YOU_HAVE_ACTIVE_CLAIM_TITLE,
+                )
+                is DeleteAccountUiState.Success.CanDelete -> error("Invalid state")
```

Comment:
Question: could DeleteAccountUiState.Success.CanDelete be separate from the others? E.g, could AlreadyRequestedDeletion, HasActiveInsurance, HasOngoingClaim be not DeleteAccountUiState.Success, but smth like DeleteAccountUiState.DeleteNotAllowed? Feels like CanDelete is very different from the other ones

Reply:
I completely agree.
Resolved with https://github.com/HedvigInsurance/android/commit/d5db1776ed25455781ff0d2c8156c5f227ff2ae1